src/System.Runtime.Serialization.Xml/tests/Performance/PerformanceTestsCommon.cs

```diff
@@ -0,0 +1,207 @@
+            {
+                IntProperty = index,
+                StringProperty = index + " string value",
+                EnumProperty = (MyEnum)(index % (Enum.GetNames(typeof(MyEnum)).Length)),
+                CollectionProperty = new List<string>(),
+                SimpleTypeList = new List<SimpleTypeWihtMoreProperties>()
+            };
+            for (int i = 0; i < collectionSize; ++i)
+            {
+                obj.CollectionProperty.Add(index + "." + i);
+            }
+            if (height > 1)
+            {
+                for (int i = 0; i < childListSize; ++i)
+                {
+                    obj.SimpleTypeList.Add(CreateObject(height - 1, index, i, collectionSize, childListSize));
+                }
+            }
+            return obj;
+        }
+
+        public static byte[] CreateByteArray(int size)
+        {
+            byte[] obj = new byte[size];
+            for (int i = 0; i < obj.Length; ++i)
+            {
+                obj[i] = (byte)(i % 256);
+            }
+            return obj;
+        }
+
+        public static Dictionary<int, string> CreateDictionaryOfIntString(int count)
+        {
+            Dictionary<int, string> dictOfIntString = new Dictionary<int, string>();
+            for (int i = 0; i < count; ++i)
+            {
+                dictOfIntString[i] = i.ToString();
+            }
+
+            return dictOfIntString;
+        }
+
+        public static List<int> CreateListOfInt(int count)
+        {
+            return Enumerable.Range(0, count).ToList();
+        }
+
+        public static object GetSerializationObject(TestType testType, int testSize)
+        {
+            Object obj = null;
+            switch (testType)
+            {
+                case TestType.ByteArray:
+                    obj = CreateByteArray(testSize);
+                    break;
+                case TestType.Dictionary:
+                    obj = CreateDictionaryOfIntString(testSize);
+                    break;
+                case TestType.ISerializable:
+                    obj = new ClassImplementingIXmlSerialiable() { StringValue = "Hello world" };
+                    break;
+                case TestType.ListOfInt:
+                    obj = CreateListOfInt(testSize);
+                    break;
+                case TestType.SimpleType:
+                    obj = CreateObject(testSize, 0, -1, 7, 2);
+                    break;
+                case TestType.String:
+                    obj = new string('k', testSize);
+                    break;
+                case TestType.XmlElement:
+                    XmlDocument xmlDoc = new XmlDocument();
+                    xmlDoc.LoadXml(@"<html></html>");
+                    XmlElement xmlElement = xmlDoc.CreateElement("Element");
+                    xmlElement.InnerText = "Element innertext";
+                    obj = xmlElement;
+                    break;
+            }
```

Comment:
In the future, we may add new test type. If we forgot to add a case entry for the new test type, this method would return null. Do we want to add an Assert or throw exception in default if the TestType is not recognized?